tests/DecoratorTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Fluent\Test;
+
+use function Fluent\create;
+use function Fluent\get;
+use Fluent\PhpConfigLoader;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\Compiler\DecoratorServicePass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class DecoratorTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function services_can_decorate_an_existing_service() {
+        $decorated = new class() {};
+        $decorating = new class(null) {
+            public function __construct($argument) {
+                $this->argument = $argument;
+            }
+        };
+        $decoratedClassName = get_class($decorated);
+        $decoratingClassName = get_class($decorating);
+
+        $container = new ContainerBuilder;
+        (new PhpConfigLoader($container))->load([
+            'decorated' => create($decoratedClassName),
+            'decorating' => create($decoratingClassName)
+                ->decorate('decorated')
+                ->arguments(get('decorating.inner')),
+        ]);
+        $this->process($container);
```

Comment:
How about calling `$container->compile()` instead? It should do the trick I think?